UI/NpcList/NpcList.gd

```diff
@@ -6,21 +6,95 @@ onready var nameButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/Name
 onready var genderButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/Gender
 onready var personalityButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/Personality
 onready var childrenButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/ChildrenAmount
+onready var popupWindow = $CenterContainer/Notification
+onready var popupWindowLabel = $CenterContainer/Notification/NotificationLabel
+onready var popupForgetBtn = $CenterContainer/Notification/HBoxC/Forget
+onready var popupOkBtn = $CenterContainer/Notification/HBoxC/Ok
+onready var popupCancelBtn = $CenterContainer/Notification/HBoxC/Cancel
 var _nameBtnState: bool = true
 var _genderBtnState: bool = true
 var _personalityBtnState: bool = true
 var _childrenBtnState: bool = true
+var _IDtoForget
+var nodeToFree
 
 
-func addRow(name: String, gender: String, subbyStat: float, ID: String, children: int = 0):
+func addRow(name: String, gender: String, subbyStat: float, ID: String, occupation: String, children: int = 0):
 	var newRow = npcRow.instance()
 	container.add_child(newRow)
-	newRow.initData(name, gender, subbyStat, ID, children)
+	newRow.initData(name, gender, subbyStat, ID, occupation, children)
 	newRow.connect("onForgetButtonPressed", self, "forgetNPC")
+	newRow.connect("onMeetButtonPressed", self, "meetNPC")
+	
+
+func forgetNPC(ID, name, node):
+	_IDtoForget = ID
+	nodeToFree = node
+	sendPopupMessage("Are you sure you want to forget "+name+"? This will permanently remove "+name+" from the game. Keep in mind that if this character is pregnant, their pregnancy will be forgotten too. But any kids you had together will stay.", true)
+
+
+func _on_Forget_pressed():
+	nodeToFree.queue_free()
+	nodeToFree = null
```

Comment:
Made it so the node get no reference and die in peace. Is this correct?

Reply:
yep